src/components/search-result-list/SearchResultList.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from "react";
+import { formatFilters } from "../../apps/search-result/helpers";
+import { FilterItemTerm } from "../../apps/search-result/types";
+import {
+  FacetField,
+  useSearchFacetQuery
+} from "../../core/dbc-gateway/generated/graphql";
+import { useCampaignMatchPOST } from "../../core/dpl-cms/dpl-cms";
+import {
+  CampaignMatchPOST200,
+  CampaignMatchPOSTBodyItem
+} from "../../core/dpl-cms/model";
+import { getCoverTint, isObjectEmpty } from "../../core/utils/helpers/general";
+import { Work } from "../../core/utils/types/entities";
+import Campaign from "../campaign/Campaign";
+import SearchResultListItem from "./search-result-list-item/search-result-list-item";
+
+export interface SearchResultListProps {
+  resultItems: Work[];
+  filters: { [key: string]: { [key: string]: FilterItemTerm } };
+  q: string;
+}
+
+export const allFacetFields = [
+  FacetField.MainLanguages,
+  FacetField.AccessTypes,
+  FacetField.ChildrenOrAdults,
+  FacetField.Creators,
+  FacetField.FictionNonfiction,
+  FacetField.FictionalCharacter,
+  FacetField.GenreAndForm,
+  FacetField.MaterialTypes,
+  FacetField.Subjects,
+  FacetField.WorkTypes
+];
+
+const SearchResultList: React.FC<SearchResultListProps> = ({
+  resultItems,
+  filters,
+  q
+}) => {
+  const { mutate } = useCampaignMatchPOST();
+  const [campaignData, setCampaignData] = useState<CampaignMatchPOST200 | null>(
+    null
+  );
+  const { data } = useSearchFacetQuery({
+    q: { all: q },
+    facets: allFacetFields,
+    facetLimit: 10,
+    ...(isObjectEmpty(filters)
+      ? {}
+      : { filters: { ...formatFilters(filters) } })
+  });
+
+  useEffect(() => {
+    if (data) {
+      mutate(
+        {
+          data: data.search.facets as CampaignMatchPOSTBodyItem[]
```

Comment:
I tried to define the arguments for the facets required by the campaign service to match the facets returned by the facet service. Having to use `as` here indicate there is a difference. I wonder what it is?

Reply:
Types of property 'score' are incompatible -> Type 'number | **null** | undefined' is not assignable to type 'number | undefined'.